mrf/js/56-data-fetching/mian.js

```diff
@@ -0,0 +1,99 @@
+// Define the API URL
+const apiUrl = "https://jsonplaceholder.typicode.com/";
+
+async function getAuthorPost(userID) {
+  const response = await fetch(apiUrl + `users/${userID}`);
+  const final = await response.json();
+
+  return final["name"];
+}
+
+async function getPost(id) {
+  const response = await fetch(apiUrl + `posts/${id}`);
+  const final = await response.json();
+
+  const author = await getAuthorPost(final["userId"]);
+  final["name"] = author;
+
+  return final;
+}
+
+allDivs = {};
```

Comment:
Instead of storing references, you can have a list of posts where each post is an object, and based on the list, implement the necessary changes in the DOM

Reply:
Can you give me an example?

Reply:
Let’s assume that your array of cards data is like this:
`let postsList = [
{
id: 0,
title: "title1",
body: "body1",
author: "author1"
},
{
id: 1,
title: "title2",
body: "body2",
author: "author2"
},
]`
 in which case you can write a function that receives this array, calls the addCard function, and adds the Cards to the body one by one.
But you must be careful to empty the body each time before calling the function. If you want to know why, you can see the result for yourself.